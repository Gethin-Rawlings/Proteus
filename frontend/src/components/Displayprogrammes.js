import React, {Component} from 'react';
import './displayusers.css'
import {withRouter} from "react-router-dom"

class Displayprogrammes extends Component {
  constructor(props) {
    super(props);
    this.state = {
      programmes: '[{"":""}]',
      history: this.props.history
    };
    this.handleClick = this
      .handleClick
      .bind(this);
  }
  handleClick(event) {
    const value = event.target.id;
    console.log(value)
    if (event) {
      this
        .props
        .history
        .push({
          pathname: "/ProgrammeDetails",
          state: {
            detail: value
          }
        });
    }
  }
  handleChange(event) {
    const target = event.target;
    const name = target.name;
    const value = target.value;
    this.setState({[name]: value});
  }
  createTable = (tableData) => {
    let tableLength=tableData.length
    let table = []
    for (let i = 0; i < tableLength; i++) {
        let children = []
        let row = Object.values(tableData[i])
        row.map(data => children.push(<td id={row[0]+row[1]} key={data}>{data}</td>))
        table.push(<tr key={row} onClick={this.handleClick}>{children}</tr>)
    }
    if (table.length === 1) {
        return null
    }
    else return table
  }
  render() {
    const returnData = JSON.parse(this.props.programmes)
    let tableHeaders = [];
    if (returnData.length>0){
      let headers = Object.keys(returnData[0]);
      headers.map(header => tableHeaders.push(<th key={header}>{header}</th>))
    }
    return (
    <section>
      <table className='displayusers'>
        <tbody>
            <tr>{tableHeaders}</tr>
            {this.createTable(returnData)}
        </tbody>
      </table> 
    </section>
    )
  }
}
export default withRouter(Displayprogrammes);