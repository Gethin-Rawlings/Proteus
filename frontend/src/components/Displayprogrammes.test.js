import React from 'react';
import * as Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';
import Displayprogrammes from './Displayprogrammes'
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

describe('storage', () => [localStorage, sessionStorage].map(storage => {
    beforeEach(() => {
      storage.clear();
      jest.clearAllMocks();
    });
  
    Enzyme.configure({ adapter: new ReactSixteenAdapter() });
  
    const history = ['/Displayprogrammes'];
  
    test('Displayprogrammes component should render as expected', () => {
      const KEY = 'loggedIn',
        VALUE = true;
      sessionStorage.setItem(KEY, VALUE);
      const component = shallow(<Displayprogrammes history={history} />);
      const tree = component;
      expect(tree).toMatchSnapshot();
    })
    /*
    test('It responds to a mouse click', () => {
      const wrapper = shallow(<Displayprogrammes />);
      wrapper.find('').simulate('click', { preventDefault() {} });

    })
    */
  }));