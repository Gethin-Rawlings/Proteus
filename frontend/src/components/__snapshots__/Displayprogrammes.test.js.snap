// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storage Displayprogrammes component should render as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Displayprogrammes)
    history={
        Array [
            "/Displayprogrammes",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`storage Displayprogrammes component should render as expected 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Displayprogrammes)
    history={
        Array [
            "/Displayprogrammes",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
